packages/create-plugin/src/utils/utils.templates.ts

```diff
@@ -88,40 +93,66 @@ export function renderTemplateFromFile(templateFile: string, data?: any) {
-    isAppType: pluginJson.type === PLUGIN_TYPES.app || pluginJson.type === PLUGIN_TYPES.scenes,
-    isNPM: packageManagerName === 'npm',
-    packageManagerName,
-    packageManagerVersion,
-    packageManagerInstallCmd,
-    version: currentVersion,
-    bundleGrafanaUI: features.bundleGrafanaUI ?? DEFAULT_FEATURE_FLAGS.bundleGrafanaUI,
-    useReactRouterV6: useReactRouterV6,
-    reactRouterVersion: useReactRouterV6 ? '6.22.0' : '5.2.0',
-    usePlaywright,
-    e2eTestCmd,
-  };
 
-  debug('\nTemplate data:\n' + JSON.stringify(templateData, null, 2));
+  if (data) {
+    const { packageManagerName, packageManagerVersion } = getPackageManagerFromUserAgent();
+    const packageManagerInstallCmd = getPackageManagerInstallCmd(packageManagerName);
+    const e2eTestCmd = usePlaywright
+      ? 'playwright test'
+      : `${packageManagerName} exec cypress install && ${packageManagerName} exec grafana-e2e run`;
+    const useReactRouterV6 = features.useReactRouterV6 === true && data.pluginType === PLUGIN_TYPES.app;
+    templateData = {
+      ...EXTRA_TEMPLATE_VARIABLES,
+      ...data,
+      pluginId: normalizeId(data.pluginName, data.orgName, data.pluginType),
+      packageManagerName,
+      packageManagerInstallCmd,
+      packageManagerVersion,
+      isAppType: data.pluginType === PLUGIN_TYPES.app || data.pluginType === PLUGIN_TYPES.scenes,
+      isNPM: packageManagerName === 'npm',
+      version: currentVersion,
+      bundleGrafanaUI: features.bundleGrafanaUI ?? DEFAULT_FEATURE_FLAGS.bundleGrafanaUI,
+      useReactRouterV6,
+      reactRouterVersion: useReactRouterV6 ? '6.22.0' : '5.2.0',
+      usePlaywright,
+      e2eTestCmd,
+    };
+  } else {
+    const pluginJson = getPluginJson();
+    const { packageManagerName, packageManagerVersion } = getPackageManagerWithFallback();
+    const packageManagerInstallCmd = getPackageManagerInstallCmd(packageManagerName);
+    const e2eTestCmd = usePlaywright
+      ? 'playwright test'
+      : `${packageManagerName} exec cypress install && ${packageManagerName} exec grafana-e2e run`;
+    const useReactRouterV6 = features.useReactRouterV6 === true && pluginJson.type === PLUGIN_TYPES.app;
+    templateData = {
+      ...EXTRA_TEMPLATE_VARIABLES,
+      pluginId: pluginJson.id,
+      pluginName: pluginJson.name,
+      pluginDescription: pluginJson.info?.description,
+      hasBackend: pluginJson.backend,
+      orgName: pluginJson.info?.author?.name,
+      pluginType: pluginJson.type,
+      packageManagerName,
+      packageManagerInstallCmd,
+      packageManagerVersion,
+      isAppType: pluginJson.type === PLUGIN_TYPES.app || pluginJson.type === PLUGIN_TYPES.scenes,
+      isNPM: packageManagerName === 'npm',
+      version: currentVersion,
+      bundleGrafanaUI: features.bundleGrafanaUI ?? DEFAULT_FEATURE_FLAGS.bundleGrafanaUI,
+      useReactRouterV6,
+      reactRouterVersion: useReactRouterV6 ? '6.22.0' : '5.2.0',
+      usePlaywright,
+      e2eTestCmd,
+    };
+  }
```

Comment:
Looks like there are is some redundant logic in the two branches of the if-else, that we could unify (e.g. `reactRouterVersion`, etc.).

What if we would only set a few properties of the template data using the arguments? Instead of the if-else structure here I would probably build a single object and rely on the data object argument for each specific property.